### PR TITLE
LiftoverVcf was failing on variants that have swapped alleles when the…

### PR DESCRIPTION
**Pull request by yfarjoun**
… AF field was the missing value ".".

- fixed bug,
- added tests.

### Description

_Give your PR a **concise** yet **descriptive** title_
_Please explain the changes you made here._
_Explain the **motivation** for making this change. What existing problem does the pull request solve?_
_Mention any issues fixed, addressed or otherwise related to this pull request, including issue numbers or hard links for issues in other repos._
_You can delete these instructions once you have written your PR description._

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

#### Content
- [ ] Added or modified tests to cover changes and any new functionality
- [ ] Edited the README / documentation (if applicable)
- [ ] All tests passing on Travis

#### Review
- [ ] Final thumbs-up from reviewer
- [ ] Rebase, squash and reword as applicable

For more detailed guidelines, see https://github.com/broadinstitute/picard/wiki/Guidelines-for-pull-requests

### CODE REVIEW

File: src/main/java/picard/util/LiftoverUtils.java [hunk 1]
```diff
@@ -286,15 +287,9 @@ public static VariantContext swapRefAlt(final VariantContext vc, final Collectio
...
 
-                if (attributeToReverse == -1) {
-                    log.warn("Trying to reverse attribute " + key +
-                            "but found value that isn't parsable as a double: (" + vc.getAttribute(key, "") +
-                            ") in variant " + vc + ". Results might be wrong.");
-                }
-
                 if (attributeToReverse < 0 || attributeToReverse > 1) {
                     log.warn("Trying to reverse attribute " + key +
                             "but found value that isn't between 0 and 1: (" + attributeToReverse + ") in variant " + vc + ". Results might be wrong.");
```
Comment (gbggrant) [inline src/main/java/picard/util/LiftoverUtils.java]: Not yours, but "but" should start with a leading space.

File: src/test/java/picard/util/LiftoverUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+    static final Allele A = Allele.create("A", false);
+    static final Allele T = Allele.create("T", false);
+    static final Allele C = Allele.create("C", false);
+    static final Allele RefT = Allele.create("T", true);
+    static final Allele RefA = Allele.create("A", true);
+    static final Allele RefC = Allele.create("C", true);
+    static final Allele RefG = Allele.create("G", true);
+
+    static String AF = "AF", MAX_AF = "MAX_AF";
```
Comment (gbggrant) [inline src/test/java/picard/util/LiftoverUtilsTest.java]: I prefer each declaration on its own line.  Not sure if we have a standard for this.

File: src/test/java/picard/util/LiftoverUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+ */
+public class LiftoverUtilsTest {
+
+    static final Allele A = Allele.create("A", false);
+    static final Allele T = Allele.create("T", false);
+    static final Allele C = Allele.create("C", false);
+    static final Allele RefT = Allele.create("T", true);
+    static final Allele RefA = Allele.create("A", true);
+    static final Allele RefC = Allele.create("C", true);
+    static final Allele RefG = Allele.create("G", true);
```
Comment (gbggrant) [inline src/test/java/picard/util/LiftoverUtilsTest.java]: T, RefT and RefG aren't used

File: src/test/java/picard/util/LiftoverUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+        testName = "test5";
+        builder.source(testName);
+        builder.attribute("AF_MAX_shouldnt_be_dropped", 1);
+
+        resultBuilder.source(testName);
+        resultBuilder.attribute("AF_MAX_shouldnt_be_dropped", 1);
+        tests.add(new Object[]{builder.make(), resultBuilder.make()});
+
+
```
Comment (gbggrant) [inline src/test/java/picard/util/LiftoverUtilsTest.java]: Delete some of these lines.

File: src/test/java/picard/util/LiftoverUtilsTest.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+
+
+        return tests.toArray(new Object[0][]);
+    }
+
+    @Test(dataProvider = "swapRefAltData")
+    public void testSwapRefAlt(final VariantContext swapMe, final VariantContext expected) throws Exception {
```
Comment (gbggrant) [inline src/test/java/picard/util/LiftoverUtilsTest.java]: Test doesn't throw an exception.

**Comment by yfarjoun** [comment]
fixes #1037

**Pull request merged**